src/wrappers/cobs.py

```diff
@@ -1,15 +1,106 @@
+import subprocess
+from datetime import datetime
 from os import environ
+from pathlib import Path
+from tempfile import TemporaryDirectory
 
 import cobs_index as cobs
 
-
-COBS_INDEX_PATH = environ.get('COBS_INDEX_PATH', '/data/500.cobs_compact')
+COBS_CLASSIC_INDEXES_DIR = environ.get('COBS_CLASSIC_INDEXES_DIR', '/data/classic')
+COBS_CLASSIC_FILE_EXTENSION = 'cobs_classic'
 COBS_TERM_SIZE = 31
+COBS_POSITIVE_RATE = 0.4
 
 
 class Cobs:
-    def __init__(self, index_path=COBS_INDEX_PATH):
-        self.search_instance = cobs.Search(index_path)
+    def __init__(self):
+        index_paths = []
+        for path in Path(COBS_CLASSIC_INDEXES_DIR).iterdir():
+            if path.is_dir():
+                index_paths += [p for p in path.iterdir() if Cobs.is_classic_index(p)]
+        self.search_instances = [cobs.Search(str(index_path)) for index_path in index_paths]
 
     def search(self, query, threshold):
-        return self.search_instance.search(query, threshold=threshold)
+        results = []
+
+        for search_instance in self.search_instances:
+            results += search_instance.search(query, threshold=threshold)
+
+        flatten = []
+        for r in results:
+            flatten.append((r.score, r.doc_name))
+        return flatten
+
+    @staticmethod
+    def build(sample_paths, sample_names):
+        assert len(sample_paths) == len(sample_names)
+
+        p = cobs.ClassicIndexParameters()
+        p.term_size = COBS_TERM_SIZE
+        p.false_positive_rate = COBS_POSITIVE_RATE
+        setattr(p, 'continue', True)
+
+        samples_by_sig_size = Cobs.group_samples_by_signature_size(sample_paths, sample_names, p)
+        Cobs.build_new_indexes(samples_by_sig_size, p)
+        Cobs.combine_classic_indexes(p)
+
+    @staticmethod
+    def group_samples_by_signature_size(sample_paths, sample_names, params):
+        sig_sizes = sorted([int(x.name) for x in Path(COBS_CLASSIC_INDEXES_DIR).iterdir() if x.is_dir() and x.name.isnumeric()])
+        samples_by_sig_size = {x: [] for x in sig_sizes}
+
+        # cobs.DocumentEntry doesn't have a constructor, so the only way to construct one is to iterate a DocumentList
+        with TemporaryDirectory() as tmpdir:
+            doclist = cobs.DocumentList(tmpdir)
+            for original_sample_path in sample_paths:
+                doclist.add(original_sample_path.strip())
+
+            # Copy sample file to the group with the smallest signature size that is larger than its own
+            for doc, name in zip(doclist, sample_names):
+                signature_size = cobs.calc_signature_size(doc.num_terms(COBS_TERM_SIZE), params.num_hashes, params.false_positive_rate)
+
+                for sig_size in sig_sizes:
+                    if sig_size > signature_size:
```

Comment:
Have you consider what we should do if the calculated signature size is greater than any of the signature sizes we currently have?